src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts

```diff
@@ -183,10 +289,22 @@ class encryptionSettings {
         return matched ? matched.label : null;
     }
 
-    useEncryptionType(mode: Raven.Client.Documents.Operations.Backups.EncryptionMode) {
+    setEncryptionType(mode: Raven.Client.Documents.Operations.Backups.EncryptionMode) {
         this.mode(mode);
     }
 
+    setKeyUsedBeforeSave() {
+        // Use 'original key' for 'save' when: 
+        // opened existing encrypted backup and didn't request to change key
+        // or if 'key' does exist but asked to change 
+        // remember - 'key' has no value if wasn't generated from UI...
+        if ((!this.key() && !!this.originalKey()) ||
```

Comment:
can we simply this? Also `!!` are not needed in javascript.

Reply:
changed to:

```
        if (this.originalKey() && 
           (!this.key() || (this.key() && !this.changeKeyRequest()))) {
            
            this.key(this.originalKey()); // using original
            this.keyConfirmation(true);   // needed for validation  
        }
```

Reply:
!this.key() || !this.changeKeyRequest()

Reply:
done